storage_v2/spanstore/reader.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package spanstore
+
+import (
+	"context"
+	"time"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+// Reader finds and loads traces and other data from storage.
+type Reader interface {
+	// GetTrace retrieves the trace with a given id.
+	//
+	// If no spans are stored for this trace, it returns ErrTraceNotFound.
+	GetTrace(ctx context.Context, traceID model.TraceID) (*model.Trace, error)
```

Comment:
the return value should be in ptrace model, similar to Writer

> GetTrace(ctx context.Context, traceID model.TraceID) (*model.Trace, error)

Isn't there a TraceID type in `ptrace`? I would use that - we want to do a clean break from legacy `/model/`

Reply:
> Isn't there a TraceID type in ptrace?

Yes, they have TraceID type defined in the pcommon pkg.

> we want to do a clean break from legacy `/model/`

I see, I'll keep that in mind!